.github/workflows/build.yml

```diff
@@ -240,9 +266,3 @@ jobs:
 
       - name: "PHPStan"
         run: "make phpstan"
-
-      - uses: "actions/upload-artifact@v4"
```

Comment:
instead of uploading/downloading artifacts for caching, I changed to use `actions/cache` because that way it also works on phpstan-src. special in phpstan-src is that the "static-analysis" and "test" jobs are spread across 2 workflows, which makes artifact handling complicated